src/Process/3dTilesProcessing.js

```diff
@@ -220,36 +220,28 @@ export function pre3dTilesUpdate(context, layer) {
     return [layer.root];
 }
 
-// Improved zoom geometry
+const cameraLocalPosition = new THREE.Vector3();
+const worldPosition = new THREE.Vector3();
 function computeNodeSSE(camera, node) {
+    node.distance = 0;
     if (node.boundingVolume.region) {
-        const cameraLocalPosition = camera.camera3D.position.clone();
-        cameraLocalPosition.x -= node.boundingVolume.region.matrixWorld.elements[12];
-        cameraLocalPosition.y -= node.boundingVolume.region.matrixWorld.elements[13];
-        cameraLocalPosition.z -= node.boundingVolume.region.matrixWorld.elements[14];
-        const distance = node.boundingVolume.region.box3D.distanceToPoint(cameraLocalPosition);
-        node.distance = distance;
-        return camera.preSSE * (node.geometricError / distance);
+        worldPosition.setFromMatrixPosition(node.boundingVolume.region.matrixWorld);
+        cameraLocalPosition.copy(camera.camera3D.position).sub(worldPosition);
+        node.distance = node.boundingVolume.region.box3D.distanceToPoint(cameraLocalPosition);
+    } else if (node.boundingVolume.box) {
+        worldPosition.setFromMatrixPosition(node.matrixWorld);
+        cameraLocalPosition.copy(camera.camera3D.position).sub(worldPosition);
+        node.distance = node.boundingVolume.box.distanceToPoint(cameraLocalPosition);
+    } else if (node.boundingVolume.sphere) {
+        worldPosition.setFromMatrixPosition(node.matrixWorld);
+        cameraLocalPosition.copy(camera.camera3D.position).sub(worldPosition);
+        node.distance = node.boundingVolume.sphere.distanceToPoint(cameraLocalPosition);
     }
-    if (node.boundingVolume.box) {
-        const cameraLocalPosition = camera.camera3D.position.clone();
-        cameraLocalPosition.x -= node.matrixWorld.elements[12];
-        cameraLocalPosition.y -= node.matrixWorld.elements[13];
-        cameraLocalPosition.z -= node.matrixWorld.elements[14];
-        const distance = node.boundingVolume.box.distanceToPoint(cameraLocalPosition);
-        node.distance = distance;
-        return camera.preSSE * (node.geometricError / distance);
+    if (node.distance <= 0) {
```

Comment:
> for some reason, the distanceToPoint of the box and sphere object do not behave in the same way, the spherical one can return a negative number

because `box3.distanceToPoint(pt)` clamps `pt` to box's surface 
You would do : 
```js
function computeNodeSSE(camera, node) {
    node.distance = 0;
    if (node.boundingVolume.region) {
        worldPosition.setFromMatrixPosition(node.boundingVolume.region.matrixWorld);
        cameraLocalPosition.copy(camera.camera3D.position).sub(worldPosition);
        node.distance = node.boundingVolume.region.box3D.distanceToPoint(cameraLocalPosition);
    } else if (node.boundingVolume.box) {
        worldPosition.setFromMatrixPosition(node.matrixWorld);
        cameraLocalPosition.copy(camera.camera3D.position).sub(worldPosition);
        node.distance = node.boundingVolume.box.distanceToPoint(cameraLocalPosition);
    } else if (node.boundingVolume.sphere) {
        worldPosition.setFromMatrixPosition(node.matrixWorld);
        cameraLocalPosition.copy(camera.camera3D.position).sub(worldPosition);
        // Math.min to clamp on sphere's surface 
        node.distance = Math.min(node.boundingVolume.sphere.distanceToPoint(cameraLocalPosition), 0.0);
    } else {
        return Infinity;
    }
    return camera.preSSE * (node.geometricError / node.distance);
}
```

Reply:
> because box3.distanceToPoint(pt) clamps pt to box's surface

Yes, I understand that. I was just complaining that two Three functions with the same name didn't have the same behaviour, which is a source of error. The problem is fixed with the less than zero test.

Reply:
if you apply this proposition, we clarify the problem of different behaviours and we reduce the number of tests

```js
function computeNodeSSE(camera, node) {
    node.distance = 0;
    if (node.boundingVolume.region) {
        worldPosition.setFromMatrixPosition(node.boundingVolume.region.matrixWorld);
        cameraLocalPosition.copy(camera.camera3D.position).sub(worldPosition);
        node.distance = node.boundingVolume.region.box3D.distanceToPoint(cameraLocalPosition);
    } else if (node.boundingVolume.box) {
        worldPosition.setFromMatrixPosition(node.matrixWorld);
        cameraLocalPosition.copy(camera.camera3D.position).sub(worldPosition);
        node.distance = node.boundingVolume.box.distanceToPoint(cameraLocalPosition);
    } else if (node.boundingVolume.sphere) {
        worldPosition.setFromMatrixPosition(node.matrixWorld);
        cameraLocalPosition.copy(camera.camera3D.position).sub(worldPosition);
        // Math.min to clamp on sphere's surface to avoid negative distance
        node.distance = Math.min(node.boundingVolume.sphere.distanceToPoint(cameraLocalPosition), 0.0);
    } else {
        return Infinity;
    }
    return camera.preSSE * (node.geometricError / node.distance);
}
```